plugins/catalog-backend/src/processors/UrlReaderProcessor.ts

```diff
@@ -47,14 +48,25 @@ export class UrlReaderProcessor implements CatalogProcessor {
   // This limiter is used for only consuming a limited number of read streams
   // concurrently.
   #limiter: Limit;
+  #useUrlReadersSearch: boolean;
 
   constructor(
     private readonly options: {
       reader: UrlReaderService;
       logger: LoggerService;
+      config?: Config;
     },
   ) {
     this.#limiter = limiterFactory(5);
+
+    this.#useUrlReadersSearch =
+      this.options.config?.getOptionalBoolean('catalog.useUrlReadersSearch') ||
+      false;
+    if (!this.#useUrlReadersSearch) {
+      this.options.logger.warn(
+        'UrlReaderProcessor uses the legacy readUrl/search behavior which will be removed in a future release. Set catalog.useUrlReadersSearch to true to adopt the new behavior.',
```

Comment:
👍 

Given how strong of a recommendation this is, let's update the `app-config.yaml` that ships with `@backstage/create-app` to set this flag as well, otherwise we ship with this warning out of the box:

https://github.com/backstage/backstage/blob/612817b0ce5cd945cc7a9ca16a73f478d8aa296c/packages/create-app/templates/default-app/app-config.yaml.hbs#L74

Reply:
Done 👍 

I added a small comment to give better context on the option